scripts/silo/common.py

```diff
@@ -0,0 +1,72 @@
+import os
+import re
+from typing import Iterable, Tuple
+
+
+def apply_decorators(
+    decorator_name: str,
+    import_stmt: str,
+    target_names: Iterable[Tuple[str, str]],
+) -> None:
+    def find_source_paths():
+        for (dirpath, dirnames, filenames) in os.walk("./src/sentry"):
+            for filename in filenames:
+                if filename.endswith(".py"):
+                    yield os.path.join(dirpath, filename)
+
+    def find_class_declarations():
+        for src_path in find_source_paths():
+            with open(src_path) as f:
+                src_code = f.read()
+            for match in re.findall(r"\nclass\s+(\w+)\(", src_code):
+                yield src_path, match
+
+    def insert_import(src_code: str) -> str:
+        future_import = None
+        for future_import in re.finditer(r"from\s+__future__\s+.*\n+", src_code):
+            pass  # iterate to last match
+        if future_import:
+            start, end = future_import.span()
+            return src_code[:end] + import_stmt + "\n" + src_code[end:]
+        else:
+            return import_stmt + "\n" + src_code
```

Comment:
Could the import be added multiple times per file if there are several classes to decorate? I know it's not handled here, but this also applies to test classes.

Reply:
Yes. I've been relying on the `isort` pre-commit action to clean it up.